invokeai/app/invocations/sdxl.py

```diff
@@ -537,27 +538,28 @@ def invoke(self, context: InvocationContext) -> LatentsOutput:
             scheduler_name=self.scheduler,
         )
 
-        # apply denoising_start
-        num_inference_steps = self.steps
-        scheduler.set_timesteps(num_inference_steps)
-
-        t_start = int(round(self.denoising_start * num_inference_steps))
-        timesteps = scheduler.timesteps[t_start * scheduler.order :]
-        num_inference_steps = num_inference_steps - t_start
-
-        # apply noise(if provided)
-        if self.noise is not None and timesteps.shape[0] > 0:
-            noise = context.services.latents.get(self.noise.latents_name)
-            latents = scheduler.add_noise(latents, noise, timesteps[:1])
-            del noise
-
         unet_info = context.services.model_manager.get_model(
             **self.unet.unet.dict(),
             context=context,
         )
+
         do_classifier_free_guidance = True
         cross_attention_kwargs = None
         with unet_info as unet:
+            # apply denoising_start
+            num_inference_steps = self.steps
+            scheduler.set_timesteps(num_inference_steps, device=self.scheduler.device)
```

Comment:
Should this be `device=unet.device`? `self.scheduler` is a string and `self.scheduler.device` is blowing up for me, although I may have applied the patch incorrectly

Reply:
Yes! Missed this one from a previous test, thanks for testing it out 🙌 (fix just pushed)